lib/app.rb

```diff
@@ -67,7 +70,7 @@ def create_teacher
     name = gets.chomp
     print 'Specialization: '
     specialization = gets.chomp
-    teacher = Teacher.new(age: age, name: name, specialization: specialization)
+    teacher = @person_factory.create_teacher(age, name, specialization)
     @people.push(teacher)
     puts 'Teacher created successfully.'
   end
```

Comment:
- [ ] All codes responsible for receiving inputs or codes responsible for validating the input from the user should be put in the main.rb file or a separate class or class method or module. The other methods for creating a class instance or listing instances should only create those instances. 👍 😎 

---

For Example, the `create_book` method should only create a new instance of the `Book` class while codes for getting the `title` and `author` of the _**book**_  from the console app should be put in a separate method, class or module. 💻 

--- 

Kindly apply these changes for the following methods in the `lib/app.rb` file:

- `create_person`
- `create_student`
- `create_teacher`
- `create_book`
- `create_rental`
- `rental_list`

---

#### N.B: Please ensure that the app works as it should after refactoring the code. 👍